pkg/controller/cyclenoderequest/transitioner/transitions.go

```diff
@@ -61,21 +61,13 @@ func (t *CycleNodeRequestTransitioner) transitionUndefined() (reconcile.Result,
 //  2. describes the node group and checks that the number of instances in the node group matches the number we
 //     are planning on terminating
 func (t *CycleNodeRequestTransitioner) transitionPending() (reconcile.Result, error) {
```

Comment:
General note: I'm worried we're making the pending transition a bit too thick with logic. It would be nice if, longer term, we can start to add more states to deal with fixing up a CNR. We could transition from pending -> the state where we are fixing node related problems -> back to pending.

We could also just work on collapsing the total vertical space used by the function and see how we feel afterwards. To me, the function is a bit longer than acceptable at this point. I wouldn't fix this up in this particular PR though.

Reply:
I agree with you, we should do it but not in this PR. I did toy with doing it that way but I found it would be a bigger change than I wanted.

Reply:
I agree that change should happen in its own refactoring PR.